// react-accordion-dropdown/lib/components/accordion-panel.js
'use strict';

exports.__esModule = true;
exports.default = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _accordionContent = require('./accordion-content');

var _accordionContent2 = _interopRequireDefault(_accordionContent);

var _accordionHeader = require('./accordion-header');

var _accordionHeader2 = _interopRequireDefault(_accordionHeader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var AccordionPanel = function (_Component) {
  _inherits(AccordionPanel, _Component);

  function AccordionPanel(props) {
    _classCallCheck(this, AccordionPanel);

    var _this = _possibleConstructorReturn(this, _Component.call(this, props));

    _this.state = {
      checked: false
    };

    _this.handleAccordionHeaderClick = _this.handleAccordionHeaderClick.bind(_this);
    return _this;
  }

  AccordionPanel.prototype.handleAccordionHeaderClick = function handleAccordionHeaderClick() {
    this.setState({ checked: !this.state.checked });
  };

  AccordionPanel.prototype.render = function render() {
    return _react2.default.createElement(
      'div',
      { className: '' + (this.state.checked ? 'opened' : ''), id: this.props.id },
      _react2.default.createElement(_accordionHeader2.default, { content: this.props.header, headerClick: this.handleAccordionHeaderClick, arrow: this.props.arrow }),
      _react2.default.createElement(_accordionContent2.default, { content: this.props.content, isExpanded: this.state.checked })
    );
  };

  return AccordionPanel;
}(_react.Component);

exports.default = AccordionPanel;
AccordionPanel.propTypes = process.env.NODE_ENV !== "production" ? {
  header: _propTypes2.default.any.isRequired,
  content: _propTypes2.default.any.isRequired
} : {};
module.exports = exports['default'];
