// react-accordion-dropdown/lib/components/accordion.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Accordion = function Accordion(props) {

  var childrenWithProps = props.children.map(function (child, index) {
    return _react2.default.cloneElement(child, {
      id: props.accordionId,
      arrow: props.arrow,
      key: props.accordionId + '_panel_' + index
    });
  });

  return _react2.default.createElement(
    'div',
    {
      className: 'accordion',
      role: 'tablist'
    },
    childrenWithProps
  );
};

Accordion.defaultProps = {
  arrow: 'left'
};

Accordion.propTypes = process.env.NODE_ENV !== "production" ? {
  accordionId: _propTypes2.default.string.isRequired,
  arrow: _propTypes2.default.oneOf(['left', 'right', 'none'])
} : {};

exports.default = Accordion;
module.exports = exports['default'];
